open_spiel/python/examples/leduc_nfsp.py

```diff
@@ -67,120 +71,330 @@
+            os.makedirs(FLAGS.checkpoint_dir, exist_ok=True)
+            for agent in agents:
+                if agent.has_checkpoint(latest_checkpoint_dir()):
+                    agent.restore(latest_checkpoint_dir())
+                    start_episode = list_saved_checkpoints()[-1]
+
+        for ep in range(start_episode, start_episode + FLAGS.num_train_episodes):
+            if (ep + 1) % FLAGS.eval_every == 0:
+                losses = [agent.loss for agent in agents]
+                logging.info("Losses: %s", losses)
+
+                if FLAGS.use_checkpoints:
+                    for agent in agents:
+                        agent.save(latest_checkpoint_dir())
+                        agent.save(checkpoint_dir(ep))
+
+                if FLAGS.evaluation_metric == "exploitability":
+                    # Avg exploitability is implemented only for 2 players constant-sum
+                    # games, use nash_conv otherwise.
+                    expl = exploitability.exploitability(env.game, joint_avg_policy)
+                    logging.info("[%s] Exploitability AVG %s", ep + 1, expl)
+                elif FLAGS.evaluation_metric == "nash_conv":
+                    nash_conv = exploitability.nash_conv(env.game, joint_avg_policy)
+                    logging.info("[%s] NashConv %s", ep + 1, nash_conv)
+                elif FLAGS.evaluation_metric == "avg_return":
+                    if FLAGS.evaluation_opponent_pool == "recent":
+                        opponent_checkpoints = most_recent_checkpoints(FLAGS.evaluation_opponent_pool_size)
+                    elif FLAGS.evaluation_opponent_pool == "random":
+                        opponent_checkpoints = random_checkpoints(FLAGS.evaluation_opponent_pool_size)
+                    elif FLAGS.evaluation_opponent_pool == "stratified":
+                        opponent_checkpoints = stratified_checkpoints(FLAGS.evaluation_opponent_pool_size)
+                    avg_return = evaluate_monte_carlo(env, agents, nfsp_args, opponent_checkpoints, FLAGS.evaluation_num_samples, FLAGS.evaluation_opponent_sampling)
+                    logging.info("[%s] AvgReturn %s", ep + 1, avg_return)
+                else:
+                    raise ValueError(" ".join(("Invalid evaluation metric, choose from",
+                                            "'exploitability', 'nash_conv', 'avg_return.")))
+
+                logging.info("_____________________________________________")
+
+            time_step = env.reset()
+            while not time_step.last():
+                player_id = time_step.observations["current_player"]
+                agent_output = agents[player_id].step(time_step)
+                action_list = [agent_output.action]
+                time_step = env.step(action_list)
+
+            # Episode is over, step all agents with final info state.
+            for agent in agents:
+                agent.step(time_step)
 
 
 if __name__ == "__main__":
-  app.run(main)
+    app.run(main)
```

Comment:
We use 2-space indents. Can you please correct the formatting in all the files?

I think it is messing up the diff.